blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/redirect/RedirectFilter.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.blueocean.service.embedded.redirect;
+
+import hudson.Extension;
+import hudson.init.Initializer;
+import hudson.model.User;
+import hudson.util.PluginServletFilter;
+import jenkins.model.GlobalConfiguration;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Extension
+public class RedirectFilter implements Filter {
+    @Initializer(fatal = false)
+    public static void init() throws ServletException {
+        PluginServletFilter.addFilter(new RedirectFilter());
+    }
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        //noop
+    }
+
+    @Override
+    public void destroy() {
+        //noop
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest request1 = (HttpServletRequest) request;
+        if (request instanceof HttpServletRequest && "/".equals(request1.getPathInfo()) && request.getParameter("noDefaultRedirect") == null) {
+            String defaultUI = InterfaceOption.classic.getInterfaceId();
+            DefaultUserInterfaceGlobalConfiguration userInterfaceGlobalConfiguration = GlobalConfiguration.all().get(DefaultUserInterfaceGlobalConfiguration.class);
+            if (userInterfaceGlobalConfiguration != null) {
+                defaultUI = userInterfaceGlobalConfiguration.getInterfaceId();
+            }
+
+            User user = User.current();
+            if (user != null) {
+                DefaultUserInterfaceUserProperty property = user.getProperty(DefaultUserInterfaceUserProperty.class);
+                if (property != null && !property.getInterfaceId().equals(DefaultUserInterfaceUserProperty.system.getInterfaceId())) {
+                    defaultUI = property.getInterfaceId();
+                }
+            }
+
+            if (InterfaceOption.blueocean.getInterfaceId().equals(defaultUI)) {
```

Comment:
So I think there are a couple of :bug:s with this approach.

1)  agents / CLI in order to lookup the JNLP port perform a HEAD to `/` to get the CLI/JNLP host/port settings.[Functions.java](https://github.com/jenkinsci/jenkins/blob/006c51256e58d9b691956962d9fd624fb0c1e08b/core/src/main/java/hudson/Functions.java#L2037-L2053)  and this occurs *after* filter processing - so old agents (will not be able to connect).  OR even just looking for the `X-Jenkins` header to confirm it is actually a Jenkins instance.
[JnlpAgentEndpointResolver.java](https://github.com/jenkinsci/remoting/blob/563bff81f5639db0aad501f784a97e8af6742541/src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java#L179-L202)

2) How does this interact with the `UpgradeWizard`  I believe users will no longer ever see this if the default is Blue.

Reply:
Does the upgrade wizard show when you upgrade jenkins? I thought it was just on fresh install?

Reply:
the "setup wizzard" is triggered in a few scenarios.  One is you are a new install,  the other is the plugins json has marked a new plugin as a "new feature"

this is done by adding an "added" with the version that added this (which gives you just the plugin install wizard)  This has not yet been used in open source AFAICT